cpp/src/lance/io/reader.h

```diff
@@ -73,44 +87,54 @@ class FileReader {
   ::arrow::Result<std::shared_ptr<::arrow::Table>> ReadTable(
       const lance::format::Schema& schema) const;
 
+  /// Consolidate read by offset and read by indices parameters.
+  struct GetArrayParams {
+    GetArrayParams(int32_t offset);
```

Comment:
If indices are UInt64, should offset/length also be uint64 ?

Reply:
to be consistent, lemme change indices to Int32 then.